.github/workflows/push.yaml

```diff
@@ -194,6 +194,68 @@ jobs:
           name: macos-sparkle-update
           path: ${{ github.workspace }}/output
 
+  create-windows-update:
+    name: Create Windows Update 🥩
+    if: github.repository_owner == 'obsproject' && github.ref_type == 'tag'
+    runs-on: windows-2022
+    needs: build-project
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    defaults:
+      run:
+        shell: pwsh
+    environment:
+      name: bouf
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          path: "repo"
+          fetch-depth: 0
+          ref: ${{ github.ref }}
+
+      - name: Set Up Environment 🔧
+        id: setup
+        env:
+          BOUF_ACTION_HASH: '2b6ce0a7a1fbc11372283011cf70d70227634096b112c15867b303f1a799f122'
+        run: |
+          $channel = if ($env:GITHUB_REF_NAME -match "(beta|rc)") { "beta" } else { "stable" }
+          $shortHash = $env:GITHUB_SHA.Substring(0,9)
+          "channel=${channel}" >> $env:GITHUB_OUTPUT
+          "commitHash=${shortHash}" >> $env:GITHUB_OUTPUT
+          
+          # Ensure files in action haven't been modified
+          $folderHash = ''
+          $files = Get-ChildItem "${{ github.workspace }}\repo\.github\actions\bouf"
+          foreach ($file in $files) {
+            $folderHash += (Get-FileHash $file -Algorithm SHA256).Hash
+          }
+          # This is stupid but so is powershell
+          $stream = [IO.MemoryStream]::new([byte[]][char[]]$folderHash)
+          if ((Get-FileHash -InputStream $stream -Algorithm SHA256).Hash -ne "$env:BOUF_ACTION_HASH") {
+            throw "bouf action folder hash does not match."
+          }
```

Comment:
Could you walk me through the intent here?

Reply:
Ensure that all files in the `.github/actions/bouf` folder match an expected hash to verify they have not been modified. For that purpose they are hashed, those hashes are concatenated, and the hash of the concatenated hash is verified against a known one. That's a little simpler than maintaining a file or environment variables with hashes for each file individually.

Reply:
Ah, so it's the "hash of the concatenated hash" that is why you needed a MemoryStream. Okay.

Reply:
Unfortunately, despite https://github.com/PowerShell/PowerShell-RFC/issues/65 being merged over 6 years ago PowerShell still has no native means of hashing a string.